sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java

```diff
@@ -142,8 +139,8 @@ public String getZookeeperConnection() {
     }
 
     @Override
-    public ReplacementFailurePolicy getReplacementFailurePolicy() {
-        return replacementFailurePolicy;
+    public Optional<ReplacementFailurePolicy> getReplacementFailurePolicy() {
```

Comment:
Just in case, may be fine as-is: Does the constructor need an update as well? (for de/reserialization)

If there isn't one already, there should probably be a spec exercised by the tests that contains a replacement policy, specifically exercised via the reserialize operation in `BaseServiceSpecTest` => `DefaultServiceSpec.getFactory()`